src/database/DataTypes/StocktakeBatch.js

```diff
@@ -1,12 +1,21 @@
 import Realm from 'realm';
+import { createRecord } from '../utilities';
 
 export class StocktakeBatch extends Realm.Object {
+
+  destructor(database) {
+    // Delete ItemBatch that was created as a result of creating this StocktakeBatch
+    if (this.snapshotNumberOfPacks === 0 && this.itemBatch.numberOfPacks === 0) {
+      database.delete('ItemBatch', this.itemBatch);
+    }
+  }
+
   get snapshotTotalQuantity() {
     return this.snapshotNumberOfPacks * this.packSize;
   }
 
   get countedTotalQuantity() {
-    return this.countedNumberOfPacks ? this.countedNumberOfPacks * this.packSize : null;
+    return this.countedNumberOfPacks * this.packSize;
```

Comment:
What if countedNumberOfPacks is null (being that it is optional in schema)?

Reply:
For all newly created StocktakeBatches and StocktakeBatches in finalised Stocktake, countedNumberOfPacks will not be null. And migration deals with StocktakeBatches that are not finalised.  This leave one more scenario, if there is an unfinalised Stocktake, and instead of migrating, the app is uninstalled and then new version is installed (will add a case for this in Incoming sync and will make StocktakeBatch.numberOfPacks required with default value = 0)

Reply:
I think you're changing away from this as discussed but if not:

Wouldn't it be more consistent to assume StocktakeBatch.countedNumberOfPacks is StocktakeBatch.snapshotNumberOfPacks, keeping the assumption made in `createRecord`?